### ISSUE TITLE
[Bug]: s3 based terraform_remote_state not support ap-south-2 region

### ISSUE THREAD

**Issue by olfway** [hashicorp/terraform#33349]
### Terraform Core Version

1.4.6

### AWS Provider Version

5.0.1

### Affected Resource(s)

* terraform_remote_state

### Expected Behavior

Terraform should be able to read s3 based remote state in ap-south-2 region

### Actual Behavior

Terraform plan return invalid region error

### Relevant Error/Panic Output Snippet

```shell
❯ terraform plan
data.terraform_remote_state.test: Reading...

Planning failed. Terraform encountered an error while generating this plan.
╷
│ Error: Invalid AWS Region: ap-south-2

```

### Terraform Configuration Files

```terraform
provider "aws" {
  profile = "india-hyd"
  region  = "ap-south-2"
}

data "terraform_remote_state" "test" {
  backend = "s3"
  config = {
    bucket = "test"
    key    = "test.tfstate"
  }
}
```

### Steps to Reproduce

Run terraform plan with config above

It working if I add
```
skip_region_validation = true
```

### Would you like to implement a fix?

None

**Comment by amoldesai1993** [comment]
anyone working on this issue??

**Comment by yushao2** [comment]
Ok so I have spent some time to trace the issue and it seems to be due to this dependency: 

https://github.com/hashicorp/terraform/blob/main/go.mod#L31C1-L32C1

--> this is pinned to an old version `v0.7.1` where it is reading regions from `aws/aws-sdk-go` which is further pinned to an older version where the region is not updated. https://github.com/hashicorp/aws-sdk-go-base/blob/v0.7.1/go.mod#L4

https://github.com/aws/aws-sdk-go/blob/v1.31.9/aws/endpoints/defaults.go#L19


The latest stable version seems to be v1.1.0 but it is still pointing to v1.31.9 which would still have the same issues (https://github.com/hashicorp/aws-sdk-go-base/blob/v1.1.0/go.mod)

v2 is still in beta -- theoretically upgrading the go.mod to v2 beta this should be fixed but I'm not too sure what the impact is at the moment

**Comment by yushao2** [comment]
It looks only this module is ONLY used by the s3 backend component -- I am compelled to bump up the version of the module

![image](https://github.com/hashicorp/terraform/assets/36848472/1d5964b3-d62f-4004-bec9-db514d7408bd)

**Comment by gdavison** [comment]
In the interim, the workaround is to set the [backend config parameter `skip_region_validation `](https://developer.hashicorp.com/terraform/language/settings/backends/s3#skip_region_validation) to `true`.

**Comment by gdavison** [comment]
This will be addressed in Terraform v1.6. If you still see this error when v1.6 is released (or you try any of the v1.6 beta releases), please open a new issue
